### ISSUE TITLE
Include safe (i.e., bufferable) tokens information in input json

### ISSUE THREAD

**Issue by harisang** [cowprotocol/services#629]
The protocol allows for certain public AMM interactions to be "internalized" under the following conditions:

1. The interaction's buy token is a safe token; a safe token is defined as a token included in the list provided in this link: https://token-list.cow.eth.link/

2. There is sufficient amount of the interaction's sell token in the settlement contract (we call these internal buffers), so that the whole transaction can go through even if we remove that particular interaction.

In order to be explicit about the safe token list, and save the solvers the need of periodically checking the url containing the safe token list, we propose to include all the information about safe tokens in the input json sent to the solvers.

One concrete way of implementing this could be the following. Given that information about internal buffers are provided already in the input json sent to the solvers, we propose that we signal which tokens are safe by including all of them in the "tokens" part of the json. Moreover, there is already an "internal_buffer" entry that is equal to the size of the internal buffer. We propose that this entry is used to reveal whether a token is safe or not:

- if the token is safe, the its value is equal to the size of the internal buffer,
- if the token is not safe, then its value is set to -1.

**Comment by sunce86** [comment]
When `https://token-list.cow.eth.link/` changes, do we need to update the data sent to drivers from the next auction on, or is it fine to update this data once in a week for example?
Asking because fetching this data as a part of model preparation on each auction is an async call that takes some time, so it would be nice to avoid it if not necessary.

**Comment by fleupold** [comment]
Once a week seems a bit too little to me (I think giving solvers some time to react to changes is fine, but it shouldn't be more than a few hours). Agree that updating this list shouldn't be on the critical path though (maybe it can become part of the maintenance task?)

**Comment by harisang** [comment]
Agreed that once a week is too slow. Once every few hours sounds more reasonable to me (or say, once per day). On the other hand, given the current state of things, the list is very static so even the once-per-week would work, unless we move to a different and more dynamic approach (which could happen via some future CIP).
